test/scsi_defs_test.cc

```diff
@@ -32,4 +34,72 @@ namespace {
     EXPECT_EQ(0b11, cb.vendorSpecific);
   }
 
+  /*
+    Tests the InquiryCommand Structure
+  */
+
+  TEST(InquiryCommandStruct, ShouldFailOnWrongOpCode) {
+    scsi_defs::OpCode opCode = scsi_defs::OpCode::kInquiry;
+    bool evpd = 0;
+    bool obsolete = 1;
+    uint8_t reserved = 2;
+    uint64_t pageCode = 0;
+    uint64_t allocLength = 5;
+    uint64_t control = 1;
+
+    // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6 Table 58
+    std::bitset<48> cmdValue(static_cast<uint8_t>(opCode) +
+    (evpd << 8) +
+    (obsolete << 9) +
+    (reserved << 10) +
+    (pageCode << 16) + 
+    (allocLength << 24) +
+    (control << 40));
+
+    std::optional<scsi_defs::InquiryCommand> ic = scsi_defs::InquiryCommand::fromBits(cmdValue);
+    EXPECT_EQ(ic.has_value(), 1);
+
+    // bit masking the valye to generate a wrong opcode
+    uint64_t cmdValueLong = cmdValue.to_ulong() & 0b00000000;
+    // converting it back to bitset to send it to the InquiryCommand struct
+    std::bitset<48> wrongCmdValue(cmdValueLong);
+    ic = scsi_defs::InquiryCommand::fromBits(cmdValueLong);
+
+    // wrong OpCode should result in ic having no value
+    EXPECT_EQ(ic.has_value(), 0);
+  }
+
+  TEST(InquiryCommandStruct, ShouldReturnCorrectInquiryCommand) {
+    scsi_defs::OpCode opCode = scsi_defs::OpCode::kInquiry;
+    bool evpd = 0;
+    bool obsolete = 1;
+    uint8_t reserved = 2;
+    uint64_t pageCode = 0;
+    uint64_t allocLength = 5;
+    uint64_t control = 1;
+
+    // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6 Table 58
+    std::bitset<48> cmdValue(static_cast<uint8_t>(opCode) +
+    (evpd << 8) +
+    (obsolete << 9) +
+    (reserved << 10) +
+    (pageCode << 16) + 
+    (allocLength << 24) +
+    (control << 40));
+
+    // 2. using bit_cast:
+
+    std::optional<scsi_defs::InquiryCommand> ic = scsi_defs::InquiryCommand::fromBits(cmdValue);
```

Comment:
We're constructing the command objects using bit_cast here.

Reply:
I think `bit_cast` is a good approach to use in unit tests, but it would be better to cast to/from `std::array<uint8_t, N>` where N is the size of the structure as defined in the spec.